go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -324,6 +325,180 @@ func (oc *Controller) deleteEgressFirewallRules(externalID string) error {
+			}
+		}
+	}
+}
+
+// deleteDefaultAllowACLsForEgressFirewall deletes the default allow ACLs as follows:
+// - allow cluster subnet on every nodes' switch
+// - allows service subnet on every nodes' switch
+// - all allow node IPs on every nodes' switch
+func (oc *Controller) deleteDefaultAllowACLsForEgressFirewall() {
+	matchConditions := []string{}
+	v4ClusterSubnet, v6ClusterSubnet := getClusterSubnets()
+	v4ServiceSubnet, v6ServiceSubnet := getServiceSubnets()
+	if config.Gateway.Mode == config.GatewayModeLocal {
+		matchConditions = append(matchConditions, generateDefaultClusterSubnetAllowMatch(v4ClusterSubnet, v6ClusterSubnet)...)
+		matchConditions = append(matchConditions, generateDefaultServiceSubnetAllowMatch(v4ServiceSubnet, v6ServiceSubnet, v4ClusterSubnet, v6ClusterSubnet)...)
+	}
+	nodes, err := oc.watchFactory.GetNodes()
+	if err != nil {
+		klog.Errorf("Unable to delete default allow node ACLs, error fetching nodes, err: %v", err)
+	}
+	for _, node := range nodes {
+		matchConditions = append(matchConditions, oc.generateDefaultNodeAllowMatch(node, v4ClusterSubnet, v6ClusterSubnet)...)
+	}
+	logicalSwitches := oc.getEgressFirewallACLSwitches()
+	for _, logicalSwitch := range logicalSwitches {
+		for _, match := range matchConditions {
+			_, stderr, err := util.RunOVNNbctl(
+				"acl-del",
+				logicalSwitch,
+				types.DirectionFromLPort,
+				types.EgressFirewallDefaultInternalClusterPriority,
+				match,
+			)
+			if err != nil {
+				klog.Errorf("Unable to delete default allow ACL on switch: %s, stderr: %s, err: %v", logicalSwitch, stderr, err)
+			}
+		}
+	}
+}
+
+func generateDefaultClusterSubnetAllowMatch(v4ClusterSubnets, v6ClusterSubnets []*net.IPNet) []string {
+	matchConditions := []string{}
+	for _, v4Subnet := range v4ClusterSubnets {
+		matchConditions = append(matchConditions, fmt.Sprintf("ip4.src == %s && ip4.dst == %s", v4Subnet.String(), v4Subnet.String()))
+	}
+	for _, v6Subnet := range v6ClusterSubnets {
+		matchConditions = append(matchConditions, fmt.Sprintf("ip6.src == %s && ip6.dst == %s", v6Subnet.String(), v6Subnet.String()))
+	}
+	return matchConditions
+}
+
+func generateDefaultServiceSubnetAllowMatch(v4ServiceSubnet, v6ServiceSubnet *net.IPNet, v4ClusterSubnets, v6ClusterSubnets []*net.IPNet) []string {
+	matchConditions := []string{}
+	if v4ServiceSubnet != nil {
+		for _, v4Subnet := range v4ClusterSubnets {
+			matchConditions = append(matchConditions, fmt.Sprintf("ip4.src == %s && ip4.dst == %s", v4Subnet.String(), v4ServiceSubnet.String()))
+		}
+	}
+	if v6ServiceSubnet != nil {
+		for _, v6Subnet := range v6ClusterSubnets {
+			matchConditions = append(matchConditions, fmt.Sprintf("ip6.src == %s && ip6.dst == %s", v6Subnet.String(), v6ServiceSubnet.String()))
+		}
+	}
+	return matchConditions
+}
+
+func (oc *Controller) generateDefaultNodeAllowMatch(addNode *v1.Node, v4ClusterSubnets, v6ClusterSubnets []*net.IPNet) []string {
```

Comment:
doctext on this function is missing and I'm going to interpret this as you are trying to add an allow ACL for packets destined to a single node ip? Then the caller is either applying that to the join switch (shared gw) or node switches?

Reply:
Yes, correct. Concerning the doctext, I wrote a doctext on all functions calling this one: `addNodeForEgressFirewall` / `deleteNodeForEgressFirewall` / `deleteDefaultAllowACLsForEgressFirewall`, do you still feel that I should add a doctext here?

Reply:
please